test/performance/tests/test_performance.py

```diff
@@ -30,3 +31,91 @@ def test_load_test(instances: List[harness.Instance]):
     assert "control-plane" in util.get_local_node_status(cluster_node)
     assert "control-plane" in util.get_local_node_status(joining_node)
     assert "control-plane" in util.get_local_node_status(joining_node_2)
+
+    configure_kube_burner(cluster_node)
+    process_dict = collect_metrics(instances)
+    run_kube_burner(cluster_node)
+    stop_metrics(instances, process_dict)
+    pull_metrics(instances)
+
+
+def test_single_node_load(session_instance: harness.Instance):
+    """Test the performance of a single node cluster with all features enabled."""
+    configure_kube_burner(session_instance)
+    process_dict = collect_metrics([session_instance])
+    run_kube_burner(session_instance)
+    stop_metrics([session_instance], process_dict)
+    pull_metrics([session_instance])
+
+
+def stop_metrics(instances: List[harness.Instance], process_dict: dict):
+    """Stops collecting metrics in the background from each instance."""
+    for instance in instances:
+        process_dict[instance.id].kill()
+
+
+def collect_metrics(instances: List[harness.Instance]):
+    """
+    Starts collecting metrics in the background from each instance. Returns a dictionary
+    with the process object for each instance.
+    """
+    process_dict = {}
+    for instance in instances:
+        pid = instance.exec(
+            ["pgrep", "k8s-dqlite"], text=True, capture_output=True
+        ).stdout.strip()
+        instance.exec(["apt-get", "install", "-y", "sysstat"])
+        subprocess = instance.exec_with_popen(
+            [
+                "pidstat",
+                "-druh",
+                "-p",
+                pid,
+                "1",
+                ">",
+                f"/root/{instance.id}_metrics.log",
```

Comment:
nit: Maybe we can extract this naming convention since they're being used in multiple functions.

Reply:
Since the filename is dynamic, this is likely going to add unnecessary complexity to the code.
In part 3, this filename only shows up twice- we should be fine.